wagtail/snippets/side_panels.py

```diff
@@ -0,0 +1,58 @@
+from django.forms import Media
+from django.utils.functional import cached_property
+from django.utils.translation import gettext_lazy
+
+from wagtail.admin.ui.components import Component
+
+
+class BaseSidePanel(Component):
+    def get_context_data(self, parent_context, inherit=None):
+        context = {"panel": self}
+
+        # Gather necessary context data from parent_context
+        inherit = inherit or []
+        inherit += [
+            "object",
+            "model_opts",
+            "view",
+        ]
+        for key in inherit:
+            context[key] = parent_context.get(key)
+        return context
```

Comment:
Since there's a lot of context data to be passed from the parent, I think this is a cleaner approach rather than setting each variable as an instance attribute during the side panel's instantiation from the view and then including them in the panel's context data. That approach also means that we have to instantiate the side panel in the view's `get_context_data()` (instead of something like `get_side_panels()`), since some of the variables are calculated in the generic view's `get_context_data()`.

This approach also still lets you explicitly state which context variables are needed by the panels (as opposed to just using the `parent_context` directly).